File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()
	var n, cnt int
	var s string
	fmt.Fscanf(reader, "%d\n", &n)
	m := make(map[string]bool)
	cnt = 0
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%s\n", &s)
		if s == "ENTER" {
			m = make(map[string]bool)
		} else {
			if _, f := m[s]; !f {
				m[s] = true
				cnt++
			}
		}
	}
	fmt.Fprintf(writer, "%d\n", cnt)
}

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)

n=int(input())
cnt = 0
m = dict()
for i in range(n):
    s = input().strip()
    if s=='ENTER':
        m = dict()
    else:
        if not m.get(s):
            m[s]=True
            cnt+=1
print(cnt)